lib/protocols/socket-io.js

```diff
@@ -0,0 +1,139 @@
+"use strict";
+
+var io = require("socket.io-client");
+
+var _ = require("./../utils");
+
+function property(fn) {
+  return { get: fn, configurable: true, enumerable: true };
+}
+
+/**
+ * Creates a new SocketIO Protocol instance
+ *
+ * @constructor SocketIO
+ * @param {Object} opts - protocol options
+ */
+var SocketIO = module.exports = function SocketIO(opts) {
+  this.socket = null;
+  this.connected = false;
+  this.data = null;
+  this.connections = {};
+
+  this.host = opts.host;
+  this.port = opts.port;
+
+  if (!this.host) {
+    throw new Error("No host supplied for SocketIO protocol");
+  }
+
+  if (!this.port) {
+    throw new Error("No port supplied for SocketIO protocol");
+  }
+
+  Object.defineProperties(this, {
+    url: property(function() {
+      return "http://" + this.host + ":" + this.port;
```

Comment:
WebSocket connections upgrade from plain HTTP, and Socket.IO encourages using `http` or `https` to make fallbacks easier. I was also following [Edgar's examples](https://github.com/hybridgroup/cylon-api-socketio/blob/master/examples/generic_message_event/simple-device-client.html#L29). At some point I'd also like to see if we can enable auth or https for one or more of these protocols.